src/Illuminate/Support/Collection.php

```diff
@@ -827,7 +827,12 @@ public function sortBy($callback, $options = SORT_REGULAR, $descending = false)
             $results[$key] = $this->items[$key];
         }
 
-        return new static($results);
+        // Determine if the original collection was an indexed collection and we should
+        // return an indexed collection.
+        $indexed = array_keys($this->items) === range(0, count($this->items) - 1);
+
+        return $indexed ? new static(array_values($results))
```

Comment:
Why not just, `$instance = new static($results)`, then you return either `instance`, either `$instance->values()` ?